pkg/datasource/datasource.go

```diff
@@ -119,12 +119,21 @@ func NewDatasource(
 	return ds
 }
 
-func getXrayClient(pluginContext *backend.PluginContext) (XrayClient, error) {
+type RequestSettings struct {
+	Region string
+}
+
+func getXrayClient(pluginContext *backend.PluginContext, requestSettings RequestSettings) (XrayClient, error) {
 	awsSettings, err := getDsSettings(pluginContext.DataSourceInstanceSettings)
 	if err != nil {
 		return nil, err
 	}
 
+	// add region from the request body if it's set, otherwise default region will be used
+	if requestSettings.Region != "" && requestSettings.Region != "default" {
+		awsSettings.Region = requestSettings.Region
```

Comment:
> otherwise default region will be used
Just curious, where is that happening?

Also is there a test for the default region already?

Reply:
`getDsSettings` loads in the defaults, and it was what all of them were using before. All of the pre-existing tests run for the default region